submarine-workbench/workbench-web/src/app/pages/workbench/model/model-home/model-home.component.ts

```diff
@@ -38,80 +39,23 @@ export class ModelHomeComponent implements OnInit {
-    this.modelCards = [
-      {
-        'name': "Model One",
-        'createTime': "2021-10-12",
-        'updatedTime': "2021-10-13", 
-        'tags': ["image", 'text'],
-        'description': "first model",
-      }, 
-      {
-        'name': "Model Two",
-        'createTime': "2021-10-12",
-        'updatedTime': "2021-10-13", 
-        'tags': ["speech"],
-        'description': "second model",
-      },
-      {
-        'name': "Model Three",
-        'createTime': "2021-10-12",
-        'updatedTime': "2021-10-13", 
-        'tags': ["speech"],
-        'description': "The third model has very long description for description shorten testing, 1234567890 A recurrent neural network (RNN) is a class of artificial neural networks where connections between nodes form a directed graph along a temporal sequence. This allows it to exhibit temporal dynamic behavior. Derived from feedforward neural networks, RNNs can use their internal state (memory) to process variable length sequences of inputs",
-      },
-      {
-        'name': "ModelFour",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["nlp", 'ssl'],
-        'description': "BERT, a famous ssl model in NLP.",
-      },
-      {
-        'name': "ModelFive",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["speech", 'ssl'],
-        'description': "huBERT, a famous ssl model in SP.",
-      },
-      {
-        'name': "ModelSix",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["CV", 'ssl'],
-        'description': "SimCLR, a famous ssl model in CV.",
-      },
-      {
-        'name': "ModelFour",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["nlp", 'ssl'],
-        'description': "BERT, a famous ssl model in NLP.",
-      },
-      {
-        'name': "ModelFive",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["speech", 'ssl'],
-        'description': "huBERT, a famous ssl model in SP.",
-      },
-      {
-        'name': "ModelSix",
-        'createTime': "2021-10-18",
-        'updatedTime': "2021-10-19", 
-        'tags': ["CV", 'ssl'],
-        'description': "SimCLR, a famous ssl model in CV.",
-      },
-    ];
-    
-    this.onDisplayModelCards = this.modelCards.map(card => card);
-    let tags = [];
-    this.modelCards.map((card) => {
-      Array.prototype.push.apply(tags, card.tags);
+    this.fetchModelCards();
+    this.modelService.emitInfo(null);
+  }
+  
+  fetchModelCards = () => {
+    this.modelService.fetchModelList().subscribe((res) => {
+      this.modelCards = res;
+      this.onDisplayModelCards = this.modelCards.map(card => card);
+      let tags = [];
+      this.modelCards.map((card) => {
+        Array.prototype.push.apply(tags, card.tags);
+      });
+      let tags_set = new Set(tags);
+      tags = Array.from(tags_set);
```

Comment:
We don't need to check the uniqueness of the tags again. DB has made it a primary key.

Reply:
But different models may have the same tag, right?

Reply:
I mistook of thinking this is inside the card element.But just add the element into set and then dump the set to the array will be better? (Skip one step)

Reply:
Oh, yes. This might be a better solution. I'll resolve it, thanks!

Reply:
I took a look at JavaScript Set. It turns out that JS Set does not have built-in functionality for union operation. So I think the previous implementation might be the simplest as well. 
![image](https://user-images.githubusercontent.com/59821107/143985271-5fea4c38-eef1-4568-9814-4ce0dce973bb.png)
This figure shows that if we want to union two sets, we still need to convert the sets to an array and then regenerate a new set. 
Another solution is to add the elements one by one. However, this is not more efficient and requires more code.

Reply:
`let tags_set = new Set();`
`this.modelCards.map((card) => {
        tags_set.add(card.tags);
});`
`tags = Array.from(tags_set);`
This is what I mean. Is this feasible?

Reply:
`Set.prototype.add` will treat the input as a value, so the result of the above code would be a set of arrays in it. 

An example:
![image](https://user-images.githubusercontent.com/59821107/144063577-bb091e03-3c12-405f-88fa-3d4ecc3810f2.png)